Handshake failed with fatal error SSL_ERROR_SSL

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->
```
TiKV
Release Version:   6.0.0-alpha
Edition:           Community
Git Commit Hash:   ab9bb0028e3df3bd55838acf83412c10e9964ad3
Git Commit Branch: heads/refs/tags/v6.1.0-alpha
UTC Build Time:    2022-03-18 05:25:58
Rust Version:      rustc 1.60.0-nightly (1e12aef3f 2022-02-13)
Enable Features:   jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```
### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->
```
processor	: 39
vendor_id	: GenuineIntel
cpu family	: 6
model		: 79
model name	: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
stepping	: 1
microcode	: 0xb00002a
cpu MHz		: 1861.083
cache size	: 25600 KB
physical id	: 1
siblings	: 20
core id		: 12
cpu cores	: 10
apicid		: 57
initial apicid	: 57
fpu		: yes
fpu_exception	: yes
cpuid level	: 20
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch epb cat_l3 cdp_l3 invpcid_single intel_pt ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts spec_ctrl intel_stibp
bogomips	: 4405.27
clflush size	: 64
cache_alignment	: 64
address sizes	: 46 bits physical, 48 bits virtual
power management:
```
### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->
run br_integration_test .

### What did you expect?
test passed.

### What did happened?
tikv segfault.

in tikv.log:
```
[2022/03/18 11:19:16.293 +08:00] [ERROR] [<unknown>] ["Handshake failed with fatal error SSL_ERROR_SSL: error:141CF132:SSL routines:tls_parse_ctos_key_share:bad ecpoint."]

[2022-03-18T03:19:57.852Z] [2022/03/18 11:19:16.295 +08:00] [ERROR] [<unknown>] ["assertion failed: i != pss->pollset_count"]
```

dmesg
```
[10174120.629813] grpc-server-2[2815733]: segfault at 2b ip 00005568779c6bd0 sp 00007fc9e9fcab70 error 4 in tikv-server[556875770000+30e4000]
```

the detail is in https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/10691/pipeline/.

--- a/components/tikv_util/src/timer.rs
+++ b/components/tikv_util/src/timer.rs
@@ -250,9 +250,11 @@ mod tests {
     #[test]
     fn test_global_steady_timer() {
         let t = SteadyTimer::default();
-        let timer = t.clock.now();
+        let start = t.clock.now();
         let delay = t.delay(Duration::from_millis(100));
         block_on(delay.compat()).unwrap();
-        assert!(timer.elapsed() >= Duration::from_millis(100));
+        let end = t.clock.now();
+        let elapsed = end.duration_since(start);
+        assert!(elapsed >= Duration::from_millis(100), "{:?} {:?} {:?}", start, end, elapsed);
     }
 }

--- a/components/tikv_util/src/timer.rs
+++ b/components/tikv_util/src/timer.rs
@@ -255,6 +255,12 @@ mod tests {
         block_on(delay.compat()).unwrap();
         let end = t.clock.now();
         let elapsed = end.duration_since(start);
-        assert!(elapsed >= Duration::from_millis(100), "{:?} {:?} {:?}", start, end, elapsed);
+        assert!(
+            elapsed >= Duration::from_millis(100),
+            "{:?} {:?} {:?}",
+            start,
+            end,
+            elapsed
+        );
     }
 }

--- a/tests/failpoints/cases/test_storage.rs
+++ b/tests/failpoints/cases/test_storage.rs
@@ -1343,8 +1343,8 @@ fn test_resolve_lock_deadline() {
     // Resolve lock, this needs two rounds, two process_read and two process_write.
     // So it needs more than 400ms. It will exceed the deadline.
     ctx.max_execution_duration_ms = 300;
-    fail::cfg("txn_before_process_read", "sleep(100)").unwrap();
-    fail::cfg("txn_before_process_write", "sleep(100)").unwrap();
+    fail::cfg("txn_before_process_read", "1*sleep(100)->sleep(200)").unwrap();
+    fail::cfg("txn_before_process_write", "1*sleep(100)->sleep(500)").unwrap();
     let (tx, rx) = channel();
     let mut txn_status = HashMap::default();
     txn_status.insert(TimeStamp::new(10), TimeStamp::new(0));

